projects/plugins/jetpack/modules/sitemaps/sitemap-buffer-video-xmlwriter.php

```diff
@@ -60,28 +60,28 @@ protected function initialize_buffer() {
 	 * @param array $array The URL item to append.
 	 */
 	protected function append_item( $array ) {
-		if ( ! empty( $array['url'] ) ) {
-			$this->writer->startElement( 'url' );
+		// Return early if missing fundamental data.
+		if ( empty( $array['url']['video:video'] ) ) {
```

Comment:
I note this is a slight behavior change, in that if there isn't any `video:video` data then it'll skip the `<url>` entirely now instead of outputting it like the DOMDocument generator does.

But possibly that behavior of the DOMDocument generator is incorrect. 🤷

Reply:
Yeah, I consciously made this change; if there's no video data, then it doesn't make sense to map to a page with the missing data. I could even see search engines penalizing this if they wanted.